### ISSUE TITLE
[assertions] cookie attributes case-sensitive matched

### ISSUE THREAD

**Issue by thePanz** [Orange-OpenSource/hurl#3265]
### What is the current *bug* behavior?
My server answers with the following:
```
    set-cookie: PHPSESSID=7f79194ea0bbd7f54b8a7da0a8d4d5b6; path=/; secure; httponly; samesite=lax
```

But the assertions
```
[Asserts]
cookie "PHPSESSID[Secure]" exists
cookie "PHPSESSID[HttpOnly]" exists
```

fails with:
```
   | cookie "PHPSESSID[Secure]" exists
   |   actual:   none
   |   expected: something
```

### What is the expected *correct* behavior?
Allow to match on the `secure` attribute too (case insensitive)

### Execution context
- Hurl Version (`hurl --version`):
hurl 5.0.1 (x86_64-pc-linux-gnu) libcurl/8.10.1 OpenSSL/3.3.2 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libssh2/1.11.0 nghttp2/1.63.0 quic/nghttp3/1.5.0
Features (libcurl):  alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 IDN IPv6 Largefile libz NTLM SPNEGO SSL TLS-SRP UnixSockets zstd
Features (built-in): brotli

### Possible fixes
Allow of case-insnsitive maches on the attribues? :shrug:  
Not sure if the HTTP specs require to handle the attributes in a case-sensitive way, could not find any precise documentation for that

**Comment by jcamiel** [comment]
Hi @thePanz 

According to [RFC6265](https://www.rfc-editor.org/rfc/rfc6265), attribute name should be treated case-insensitively:

>  If the attribute-name case-insensitively matches the string "Secure",
   the user agent MUST append an attribute to the cookie-attribute-list
   with an attribute-name of Secure and an empty attribute-value.

I think we should follow the spec and fix the actual behavior. Thanks for the issue!

**Comment by fabricereix** [comment]
Yes @thePanz, you are right, our cookie attribute query should be case-incentive 
to match the spec above and to be consistent with the browser.

I have tested the following response headers in Firefox
```
Set-Cookie: CamelCase=Value; Max-Age=1000; HttpOnly
Set-Cookie: lowercase=value; max-age=1000; httponly
Set-Cookie: UPPERCASE=VALUE; MAX-AGE-age=1000; HTTPONLY
```
The 3 of them have the same expiration and the HttpOnly attribute set to true.

**Comment by fabricereix** [comment]
We can also note that we have already supported the case-insensitivity of the cookie attribute in the Hurl file:

The 2 queries below are equivalent:
```
cookie "CamelCase[HttpOnly]" exists
cookie "CamelCase[httponly]" exists
```

**Comment by thePanz** [comment]
Thank you @fabricereix  and @jcamiel for the followup and the fix

Would it be possible to backport that fix to the current 5.x version of hurl?

**Comment by jcamiel** [comment]
Hi,

This fix will be integrated in the next Hurl version: in any case, backport or new version, you'll need to install a new binary. It's easier for us to just put it in the next version, we (very rarely) do patch version, only if the bug is a real show stopper.

**Comment by thePanz** [comment]
Thanks @jcamiel for the clarifications

Looking forward for the next version then, as the current one v5.0.1 (with this bug still present) is still showing errors on our usage, and we can not alter the Server's response (unfortunately)

Thanks again for the great too, BTW

**Comment by jcamiel** [comment]
Just to be sure, with 5.0.1, you have a workaround with `header` query by inspecting the `Set-Cookie` header:

```hurl
GET https://foo.com
HTTP 200
[Asserts]
header "Set-Cookie" contains "secure"
```

**Comment by thePanz** [comment]
Thanks @jcamiel ! you rock! :rocket:
